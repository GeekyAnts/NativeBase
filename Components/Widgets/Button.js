/* @flow */
'use strict';

import React from 'react';
import { TouchableOpacity, Platform } from 'react-native';
import NativeBaseComponent from '../Base/NativeBaseComponent';
import computeProps from '../../Utils/computeProps';
import IconNB from './Icon';
import Icon from './Icon';
import Text from './Text';
import _ from 'lodash';


export default class Button extends NativeBaseComponent {

    propTypes: {
        style : React.PropTypes.object,
        textStyle : React.PropTypes.object,
        block : React.PropTypes.bool,
        primary : React.PropTypes.bool,
        transparent : React.PropTypes.bool,
        success : React.PropTypes.bool,
        danger : React.PropTypes.bool,
        warning : React.PropTypes.bool,
        info : React.PropTypes.bool,
        bordered : React.PropTypes.bool,
        disabled : React.PropTypes.bool,
        rounded : React.PropTypes.bool,
        large : React.PropTypes.bool,
        small : React.PropTypes.bool,
        inputButton : React.PropTypes.bool
    }

    getInitialStyle() {
        return {
            button: {
                paddingVertical: (this.props.vertical) ? 0 :this.getTheme().buttonPadding,
                paddingHorizontal: (this.props.vertical) ? 5 :this.getTheme().buttonPadding+2,
                justifyContent: (this.props.block) ? 'center' : 'space-around',
                flexDirection:  (this.props.vertical) ? 'column' : 'row',
                alignSelf: 'center',
                alignItems: 'center',
                backgroundColor: this.getTheme().btnPrimaryBg,
                elevation: (this.props.transparent || this.props.bordered) ? 0 : 3,
                shadowColor: (this.props.transparent || this.props.bordered) ? undefined : '#000',
                shadowOffset: (this.props.transparent || this.props.bordered) ? undefined : {width: 0, height: 2},
                shadowOpacity: (this.props.transparent || this.props.bordered) ? undefined : 0.2,
                shadowRadius: (this.props.transparent || this.props.bordered) ? undefined : 2
            }
        }
    }

    prepareRootProps() {

        var type = {
            backgroundColor:    (this.props.primary) ? this.getTheme().btnPrimaryBg :
                                ((this.props.transparent) || (this.props.bordered)) ? 'rgba(0,0,0,0)' :
                                (this.props.success) ? this.getTheme().btnSuccessBg :
                                (this.props.danger) ? this.getTheme().btnDangerBg :
                                (this.props.warning) ? this.getTheme().btnWarningBg :
                                (this.props.info) ? this.getTheme().btnInfoBg :
                                (this.props.backgroundColor) ? this.props.backgroundColor :
                                (this.props.disabled) ? this.getTheme().btnDisabledBg :
                                this.getInitialStyle().button.backgroundColor,
            borderRadius: (this.props.rounded) ? this.getTheme().borderRadiusLarge : this.getTheme().borderRadiusBase,
            borderWidth: (this.props.bordered) ? 1 : 0,
            borderColor:    (this.props.primary) ? this.getTheme().primary :
                            (this.props.success) ? this.getTheme().btnSuccessBg :
                            (this.props.danger) ? this.getTheme().btnDangerBg :
                            (this.props.warning) ? this.getTheme().btnWarningBg :
                            (this.props.info) ? this.getTheme().btnInfoBg :
                            this.getInitialStyle().button.backgroundColor,
            height: (this.props.large) ? 60 : (this.props.small) ? 35 : 38,
            alignSelf: (this.props.block) ? 'stretch' : 'flex-start'
        }

        var addedProps = _.merge(this.getInitialStyle().button,type);

        var defaultProps = {
            style: addedProps,
            capitalize: true
        }

        return computeProps(this.props, defaultProps);
    }

    getTextStyle() {

        var mergedStyle = {};
        var btnType = {
            fontFamily: this.getTheme().btnFontFamily,
            marginLeft: (this.props.vertical) ? 0 : (this.iconPresent() && !this.props.iconRight) ? this.getTheme().iconMargin : 0,
            marginRight: (this.iconPresent() && this.props.iconRight) ? this.getTheme().iconMargin : 0,
            color:
                    ((this.props.bordered) && (this.props.primary)) ? this.getTheme().btnPrimaryBg :
                    ((this.props.bordered) && (this.props.success)) ? this.getTheme().btnSuccessBg :
                    ((this.props.bordered) && (this.props.danger)) ? this.getTheme().btnDangerBg :
                    ((this.props.bordered) && (this.props.warning)) ? this.getTheme().btnWarningBg :
                    ((this.props.bordered) && (this.props.info)) ? this.getTheme().btnInfoBg :
                    ((this.props.bordered)) ? this.getTheme().btnPrimaryBg :
                    (this.props.color) ? this.props.color :
                    (this.props.transparent) ? this.getContextForegroundColor() :
                    this.getTheme().inverseTextColor,

            fontSize: (this.props.large) ? this.getTheme().btnTextSizeLarge : (this.props.small) ? this.getTheme().btnTextSizeSmall : this.getTheme().btnTextSize,

            lineHeight: (this.props.large) ? 29 : (this.props.small) ? 16 : this.getTheme().btnLineHeight
        }

        return _.merge(mergedStyle, btnType, this.props.textStyle);
    }

    getIconProps(icon) {

        var defaultStyle = {
            color:
                ((this.props.bordered) && (this.props.primary)) ? this.getTheme().btnPrimaryBg :
                ((this.props.bordered) && (this.props.success)) ? this.getTheme().btnSuccessBg :
                ((this.props.bordered) && (this.props.danger)) ? this.getTheme().btnDangerBg :
                ((this.props.bordered) && (this.props.warning)) ? this.getTheme().btnWarningBg :
                ((this.props.bordered) && (this.props.info)) ? this.getTheme().btnInfoBg :
                (this.props.bordered) ? this.getTheme().btnPrimaryBg :
                (this.props.color) ? this.props.color :
                (this.props.header) ? this.getTheme().toolbarTextColor :
                (this.props.transparent) ? this.getContextForegroundColor() :
                this.getTheme().inverseTextColor,

            fontSize: (this.props.large) ? this.getTheme().iconSizeLarge : (this.props.small) ? this.getTheme().iconSizeSmall : (this.props.inputButton) ? this.getTheme().toolbarIconSize : this.getTheme().iconFontSize-5,
            lineHeight: (this.props.large) ? 52: (this.props.small || this.props.inputButton) ? 22 : this.getTheme().iconLineHeight-9
        }

        var defaultProps = {
            style: defaultStyle
        }

        return computeProps(icon.props, defaultProps);
    }
    iconPresent() {
        var iconComponentPresent = false;
        React.Children.forEach(this.props.children, function (child) {
            if(child.type == Icon)
                iconComponentPresent = true;
        })
        return iconComponentPresent;
    }
    renderChildren() {
        
        if(typeof this.props.children == 'string') {
            return <Text style={this.getTextStyle()}>{(Platform.OS==='ios' || !this.props.capitalize) ? this.props.children : this.props.children.toUpperCase()}</Text>
        }

        else if(this.props.children.type == IconNB) {
            return React.cloneElement(this.props.children, this.getIconProps(this.props.children));
        }

        else if(Array.isArray(this.props.children)) {
            var newChildren = [];

            var childrenArray = React.Children.toArray(this.props.children);

            var iconElement = [];
            iconElement = _.remove(childrenArray, function(item) {
                if(item.type == IconNB) {
                    return true;
                }
            });

            if(this.props.iconRight) {
                if (childrenArray[0].type==undefined) {
                    newChildren.push(<Text key='label' style={this.getTextStyle()}>{(Platform.OS==='ios' || !this.props.capitalize) ? childrenArray[0] : childrenArray[0].toUpperCase()}</Text>);
                } else {
                    newChildren.push(<Text key='label' style={this.getTextStyle()}>{(Platform.OS==='ios' || !this.props.capitalize) ? childrenArray[0].props.children : childrenArray[0].props.children.toUpperCase()}</Text>);
                }

                newChildren.push(<Text key='icon'>{React.cloneElement(iconElement[0], this.getIconProps(iconElement[0]))}</Text>);
            }

            else if(this.props.iconLeft || iconElement.length>0) {
                newChildren.push(<Text key='icon'>{React.cloneElement(iconElement[0], this.getIconProps(iconElement[0]))}</Text>);

                if (childrenArray[0].type==undefined) {
                    newChildren.push(<Text key='label' style={this.getTextStyle()}>{(Platform.OS==='ios' || !this.props.capitalize) ? childrenArray[0] : childrenArray[0].toUpperCase()}</Text>);
                } else {
                    newChildren.push(<Text key='label' style={this.getTextStyle()}>{(Platform.OS==='ios' || !this.props.capitalize) ? childrenArray[0].props.children : childrenArray[0].props.children.toUpperCase()}</Text>);
                }
            }
            else {
                return <Text style={this.getTextStyle()}>{(Platform.OS==='ios') ? this.props.children : this.props.children.toUpperCase()}</Text>
            }

            return newChildren;

        }

        else
            return React.cloneElement(this.props.children);

    }

    render() {
        return(
            <TouchableOpacity {...this.prepareRootProps()} >
                {this.renderChildren()}
            </TouchableOpacity>
        );
    }
}
