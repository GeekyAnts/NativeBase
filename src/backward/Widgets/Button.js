/* @flow */


import React, { Component } from 'react';
import { TouchableOpacity, Platform } from 'react-native';
import { connectStyle } from '@shoutem/theme';
import _ from 'lodash';
import variables from '../../theme/variables';
import { Icon } from './Icon';
import mapPropsToStyleNames from '../../Utils/mapPropsToStyleNames';
import computeProps from '../../Utils/computeProps';
import { Text } from './Text';


class Button extends Component {


  getInitialStyle() {
    return {
      borderedBtn: {
        borderWidth: (this.props.bordered) ? 1 : undefined,
        borderRadius: (this.props.rounded && this.props.bordered) ? variables.borderRadiusLarge : 2,
      },
    };
  }

  iconPresent() {
    let iconComponentPresent = false;
    React.Children.forEach(this.props.children, (child) => {
      if (child.type === Icon) { iconComponentPresent = true; }
    });
    return iconComponentPresent;
  }

  prepareRootProps() {
    const defaultProps = {
      style: this.getInitialStyle().borderedBtn,
    };

    return computeProps(this.props, defaultProps);
  }

  renderChildren() {
    if (typeof this.props.children === 'string') {
      return <Text style={this.props.textStyle}>{(Platform.OS === 'ios' || !this.props.capitalize) ? this.props.children : this.props.children.toUpperCase()}</Text>;
    } else if (this.props.children.type === Icon) {
      return React.cloneElement(this.props.children);
    } else if (Array.isArray(this.props.children)) {
      const newChildren = [];

      const childrenArray = React.Children.toArray(this.props.children);

      let iconElement = [];
      iconElement = _.remove(childrenArray, (item) => {
        if (item.type === Icon) {
          return true;
        }
        return null;
      });

      if (this.props.iconRight) {
        if (childrenArray[0].type === undefined) {
          newChildren.push(<Text key="label">{(Platform.OS === 'ios' || !this.props.capitalize) ? childrenArray[0] : childrenArray[0].toUpperCase()}</Text>);
        } else {
          newChildren.push(<Text key="label">{(Platform.OS === 'ios' || !this.props.capitalize) ? childrenArray[0].props.children : childrenArray[0].props.children.toUpperCase()}</Text>);
        }

        newChildren.push(React.cloneElement(iconElement[0]));
      } else if (this.props.iconLeft || iconElement.length > 0) {
        newChildren.push(React.cloneElement(iconElement[0]));

        if (childrenArray[0].type === undefined) {
          newChildren.push(<Text key="label">{(Platform.OS === 'ios' || !this.props.capitalize) ? childrenArray[0] : childrenArray[0].toUpperCase()}</Text>);
        } else {
          newChildren.push(<Text key="label">{(Platform.OS === 'ios' || !this.props.capitalize) ? childrenArray[0].props.children : childrenArray[0].props.children.toUpperCase()}</Text>);
        }
      } else {
        return <Text>{this.props.children}</Text>;
      }

      return newChildren;
    }
    return React.cloneElement(this.props.children);
  }

  render() {
    return (
      <TouchableOpacity
        ref={(c) => { this._root = c; }}
        {...this.prepareRootProps()} activeOpacity={0.5}
      >
        {this.renderChildren()}
      </TouchableOpacity>
    );
  }
}


Button.propTypes = {
  ...TouchableOpacity.propTypes,
  style: React.PropTypes.object,
  textStyle: React.PropTypes.object,
  block: React.PropTypes.bool,
  primary: React.PropTypes.bool,
  transparent: React.PropTypes.bool,
  success: React.PropTypes.bool,
  danger: React.PropTypes.bool,
  warning: React.PropTypes.bool,
  info: React.PropTypes.bool,
  bordered: React.PropTypes.bool,
  disabled: React.PropTypes.bool,
  rounded: React.PropTypes.bool,
  large: React.PropTypes.bool,
  small: React.PropTypes.bool,
  inputButton: React.PropTypes.bool,
  tabButton: React.PropTypes.bool,
};


const StyledButton = connectStyle('NativeBase.Button', {}, mapPropsToStyleNames)(Button);
export {
  StyledButton as Button,
};
