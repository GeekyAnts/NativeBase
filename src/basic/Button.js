/* @flow */


import React, { Component } from 'react';
import _ from 'lodash';
import { TouchableOpacity, Platform, View, TouchableNativeFeedback } from 'react-native';
import { connectStyle } from '@shoutem/theme';
import variables from './../theme/variables/platform';
import { Badge } from './Badge';
import { Text } from './Text';
import computeProps from '../Utils/computeProps';

import mapPropsToStyleNames from '../Utils/mapPropsToStyleNames';

class Button extends Component {

  getInitialStyle() {
    return {
      borderedBtn: {
        borderWidth: (this.props.bordered) ? 1 : undefined,
        borderRadius: (this.props.rounded && this.props.bordered) ? variables.borderRadiusLarge : 2,
      },
    };
  }

  prepareRootProps() {
    const defaultProps = {
      style: this.getInitialStyle().borderedBtn,
    };

    return computeProps(this.props, defaultProps);
  }
  renderChildren() {
    const isArray = _.isArray(this.props.children);
    if (!isArray) {
      if (this.props.children.type.displayName==="Styled(Text)") {
        return <Text {...this.props.children.props}>{(!this.props.capitalize) ? this.props.children.props.children : this.props.children.props.children.toUpperCase()}</Text>
      }
      else return this.props.children;
    }
    else {
      const newChildren = [];
      const childrenArray = _.toArray(this.props.children);
      const iconArray = _.remove(childrenArray, (child)=> child.type.displayName==="Styled(Icon)");
      if (this.props.iconRight) {
        newChildren.push(<Text key={5}>{(!this.props.capitalize) ? childrenArray[0].props.children : childrenArray[0].props.children.toUpperCase()}</Text>);
        newChildren.push(iconArray);
      } else {
        newChildren.push(iconArray);
        newChildren.push(<Text key={5}>{(!this.props.capitalize) ? childrenArray[0].props.children : childrenArray[0].props.children.toUpperCase()}</Text>);
      }
      return newChildren;
    }
  }
  render() {
    if (Platform.OS==='ios' || variables.androidRipple===false || Platform['Version'] <= 21) {
      return (
        <TouchableOpacity
          {...this.prepareRootProps()}
          ref={c => this._root = c}
          activeOpacity={(this.props.activeOpacity) ? this.props.activeOpacity : 0.5}
        >
          <View style={(Platform.OS === 'ios') ? { alignSelf: 'center', zIndex: 999 } : { alignSelf: 'center' }}>{(this.props.badgeValue) ? <Badge style={{ backgroundColor: (this.props.badgeColor) ? this.props.badgeColor : undefined }}><Text style={this.props.badgeValueStyle}>{this.props.badgeValue}</Text></Badge> : null}</View>
          {this.props.children}
        </TouchableOpacity>
      );
    }
    else {
      return(
          <TouchableNativeFeedback ref={c => this._root = c}
              onPress={this.props.onPress}
              background={(this.props.androidRippleColor) ? TouchableNativeFeedback.Ripple(this.props.androidRippleColor) : TouchableNativeFeedback.Ripple(variables.androidRippleColor)}
               {...this.prepareRootProps()}>
              <View {...this.prepareRootProps()}>
                <View style={{ alignSelf: 'center' }}>{(this.props.badgeValue) ? <Badge style={{ backgroundColor: (this.props.badgeColor) ? this.props.badgeColor : undefined }}><Text style={this.props.badgeValueStyle}>{this.props.badgeValue}</Text></Badge> : null}</View>
                  {this.renderChildren()}
              </View>
          </TouchableNativeFeedback>
      );
    }
  }
}

Button.propTypes = {
  ...TouchableOpacity.propTypes,
  style: React.PropTypes.object,
  block: React.PropTypes.bool,
  primary: React.PropTypes.bool,
  transparent: React.PropTypes.bool,
  success: React.PropTypes.bool,
  danger: React.PropTypes.bool,
  warning: React.PropTypes.bool,
  info: React.PropTypes.bool,
  bordered: React.PropTypes.bool,
  capitalize: React.PropTypes.bool,
  disabled: React.PropTypes.bool,
  rounded: React.PropTypes.bool,
  large: React.PropTypes.bool,
  small: React.PropTypes.bool,
  active: React.PropTypes.bool,
  badgeColor: React.PropTypes.string,
  badgeValueStyle: React.PropTypes.object,
};

Button.defaultProps = {
  capitalize: true
}

const StyledButton = connectStyle('NativeBase.Button', {}, mapPropsToStyleNames)(Button);
export {
  StyledButton as Button,
};
