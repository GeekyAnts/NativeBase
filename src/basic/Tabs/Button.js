const React = require("react");
const ReactNative = require("react-native");
const { TouchableOpacity, View } = ReactNative;

const Button = props => {
  return (
    <TouchableOpacity {...props}>
      {props.children}
    </TouchableOpacity>
  );
};

module.exports = Button;
//export default Button;
