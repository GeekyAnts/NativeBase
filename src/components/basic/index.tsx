export { ScrollView } from './ScrollView';
export { View } from './View';
export { StatusBar } from './StatusBar';
export { FlatList } from './FlatList';
export { SectionList } from './SectionList';
export { KeyboardAvoidingView } from './KeyboardAvoidingView';
