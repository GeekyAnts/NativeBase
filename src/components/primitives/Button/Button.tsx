import React, { memo, forwardRef } from 'react';
import Spinner from '../Spinner';
import { usePropsResolution } from '../../../hooks/useThemeProps';
import { default as Box, IBoxProps } from '../Box';
import HStack from '../Stack/HStack';
import { Pressable } from '../Pressable';
import type { IButtonProps } from './types';

const Button = (
  {
    children,
    isLoadingText,
    startIcon,
    endIcon,
    spinner,
    ...props
  }: IButtonProps & IBoxProps,
  ref: any
) => {
  const {
    _text,
    _disabled,
    _focus,
    _hover,
    _pressed,
    _focusVisible,
    _stack,
    ...resolvedProps
  } = usePropsResolution('Button', props);

  const pressableProps = {
    ...resolvedProps,
    _hover,
    _pressed,
    _focus,
    _disabled,
    _focusVisible,
  };

  const { isDisabled, isLoading } = props;

  if (endIcon && React.isValidElement(endIcon)) {
    endIcon = React.Children.map(
      endIcon,
      (child: JSX.Element, index: number) => {
        return React.cloneElement(child, {
          key: `button-end-icon-${index}`,
          ..._text,
          ...child.props,
        });
      }
    );
  }
  if (startIcon && React.isValidElement(startIcon)) {
    startIcon = React.Children.map(
      startIcon,
      (child: JSX.Element, index: number) => {
        return React.cloneElement(child, {
          key: `button-start-icon-${index}`,
          ..._text,
          ...child.props,
        });
      }
    );
  }

  return (
    <Pressable
      disabled={isDisabled || isLoading}
      ref={ref}
      {...pressableProps}
      accessibilityRole={props.accessibilityRole ?? 'button'}
    >
      {/* TODO: Replace Render props with Context Hook */}
      {/* TODO: Can look for a simpler wat to do this */}
      {({ isPressed, isHovered, isFocused }) => {
        const focusTextProps = isFocused &&
          _focus?._text && { ..._focus._text };
        const hoverTextProps = isHovered &&
          _hover?._text && { ..._hover._text };
        const pressedTextProps = isPressed &&
          _pressed?._text && { ..._pressed._text };
        const disabledTextProps = isDisabled &&
          _disabled?._text && { ..._disabled._text };
        const focusVisibleTextProps = isFocused &&
          _focusVisible?._text && { ..._focusVisible._text };

        const boxChildren =
          isLoading && isLoadingText ? isLoadingText : children;

        return (
          <HStack {..._stack}>
            {startIcon && !isLoading ? startIcon : null}
            {isLoading ? (
              spinner ? (
                spinner
              ) : (
                <Spinner color={_text?.color} size="sm" />
              )
            ) : null}
            {boxChildren ? (
              <Box
                _text={{
                  ..._text,
                  ...hoverTextProps,
                  ...focusTextProps,
                  ...focusVisibleTextProps,
                  ...pressedTextProps,
                  ...disabledTextProps,
                }}
              >
                {isLoading && isLoadingText ? isLoadingText : children}
              </Box>
            ) : null}
            {endIcon && !isLoading ? endIcon : null}
          </HStack>
        );
      }}
    </Pressable>
  );
};

export default memo(forwardRef(Button));
