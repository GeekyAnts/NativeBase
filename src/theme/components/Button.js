import { Platform } from 'react-native';
import _ from 'lodash';

import variable from './../variables/platform';


export default (variables = variable) => {
  const platformStyle = variables.platformStyle;
  const platform = variables.platform;

  const buttonTheme = {
      '.disabled': {
        backgroundColor: variables.btnDisabledBg,
      },
      '.bordered': {
        '.dark': {
          'NativeBase.Text': {
            color: '#000',
          },
          'NativeBase.Icon': {
            color: '#000',
          },
          'NativeBase.IconNB': {
            color: '#000',
          },
          backgroundColor: 'transparent',
          borderColor: '#000',
          borderWidth: variables.borderWidth * 2,
        },
        '.light': {
          'NativeBase.Text': {
            color: '#f4f4f4',
          },
          'NativeBase.Icon': {
            color: '#f4f4f4',
          },
          'NativeBase.IconNB': {
            color: '#f4f4f4',
          },
          backgroundColor: 'transparent',
          borderColor: '#f4f4f4',
          borderWidth: variables.borderWidth * 2,
        },
        '.primary': {
          'NativeBase.Text': {
            color: variables.btnPrimaryBg,
          },
          'NativeBase.Icon': {
            color: variables.btnPrimaryBg,
          },
          'NativeBase.IconNB': {
            color: variables.btnPrimaryBg,
          },
          backgroundColor: 'transparent',
          borderColor: variables.btnPrimaryBg,
          borderWidth: variables.borderWidth * 2,
        },
        '.success': {
          'NativeBase.Text': {
            color: variables.btnSuccessBg,
          },
          'NativeBase.Icon': {
            color: variables.btnSuccessBg,
          },
          'NativeBase.IconNB': {
            color: variables.btnSuccessBg,
          },
          backgroundColor: 'transparent',
          borderColor: variables.btnSuccessBg,
          borderWidth: variables.borderWidth * 2,
        },
        '.info': {
          'NativeBase.Text': {
            color: variables.btnInfoBg,
          },
          'NativeBase.Icon': {
            color: variables.btnInfoBg,
          },
          'NativeBase.IconNB': {
            color: variables.btnInfoBg,
          },
          backgroundColor: 'transparent',
          borderColor: variables.btnInfoBg,
          borderWidth: variables.borderWidth * 2,
        },
        '.warning': {
          'NativeBase.Text': {
            color: variables.btnWarningBg,
          },
          'NativeBase.Icon': {
            color: variables.btnWarningBg,
          },
          'NativeBase.IconNB': {
            color: variables.btnWarningBg,
          },
          backgroundColor: 'transparent',
          borderColor: variables.btnWarningBg,
          borderWidth: variables.borderWidth * 2,
        },
        '.danger': {
          'NativeBase.Text': {
            color: variables.btnDangerBg,
          },
          'NativeBase.Icon': {
            color: variables.btnDangerBg,
          },
          'NativeBase.IconNB': {
            color: variables.btnDangerBg,
          },
          backgroundColor: 'transparent',
          borderColor: variables.btnDangerBg,
          borderWidth: variables.borderWidth * 2,
        },
        '.disabled': {
          backgroundColor: null,
          borderColor: variables.btnDisabledBg,
          borderWidth: variables.borderWidth * 2,
          'NativeBase.Text': {
            color: variables.btnDisabledBg,
          },
        },
        'NativeBase.Text': {
          color: variables.btnPrimaryBg,
        },
        'NativeBase.Icon': {
          color: variables.btnPrimaryBg,
        },
        'NativeBase.IconNB': {
          color: variables.btnPrimaryBg,
        },
        borderWidth: variables.borderWidth * 2,
        elevation: null,
        shadowColor: null,
        shadowOffset: null,
        shadowOpacity: null,
        shadowRadius: null,
        backgroundColor: 'transparent',
      },

      '.dark': {
        '.bordered': {
          'NativeBase.Text': {
            color: '#000',
          },
          'NativeBase.Icon': {
            color: '#000',
          },
          'NativeBase.IconNB': {
            color: '#000',
          },
        },
        backgroundColor: '#000',
      },
      '.light': {
        '.transparent': {
          'NativeBase.Text': {
            color: '#f4f4f4',
          },
          'NativeBase.Icon': {
            color: '#f4f4f4',
          },
          'NativeBase.IconNB': {
            color: '#f4f4f4',
          },
          backgroundColor: null,
        },
        '.bordered': {
          'NativeBase.Text': {
            color: '#f4f4f4',
          },
          'NativeBase.Icon': {
            color: '#f4f4f4',
          },
          'NativeBase.IconNB': {
            color: '#f4f4f4',
          },
        },
        'NativeBase.Text': {
            color: '#000',
          },
          'NativeBase.Icon': {
            color: '#000',
          },
          'NativeBase.IconNB': {
            color: '#000',
          },
        backgroundColor: '#f4f4f4',
      },

      '.primary': {
        '.bordered': {
          'NativeBase.Text': {
            color: variables.btnPrimaryBg,
          },
          'NativeBase.Icon': {
            color: variables.btnPrimaryBg,
          },
          'NativeBase.IconNB': {
            color: variables.btnPrimaryBg,
          },
        },
        backgroundColor: variables.btnPrimaryBg,
      },

      '.success': {
        '.bordered': {
          'NativeBase.Text': {
            color: variables.btnSuccessBg,
          },
          'NativeBase.Icon': {
            color: variables.btnSuccessBg,
          },
          'NativeBase.IconNB': {
            color: variables.btnSuccessBg,
          },
        },
        backgroundColor: variables.btnSuccessBg,
      },

      '.info': {
        '.bordered': {
          'NativeBase.Text': {
            color: variables.btnInfoBg,
          },
          'NativeBase.Icon': {
            color: variables.btnInfoBg,
          },
          'NativeBase.IconNB': {
            color: variables.btnInfoBg,
          },
        },
        backgroundColor: variables.btnInfoBg,
      },

      '.warning': {
        '.bordered': {
          'NativeBase.Text': {
            color: variables.btnWarningBg,
          },
          'NativeBase.Icon': {
            color: variables.btnWarningBg,
          },
          'NativeBase.IconNB': {
            color: variables.btnWarningBg,
          },
        },
        backgroundColor: variables.btnWarningBg,
      },

      '.danger': {
        '.bordered': {
          'NativeBase.Text': {
            color: variables.btnDangerBg,
          },
          'NativeBase.Icon': {
            color: variables.btnDangerBg,
          },
          'NativeBase.IconNB': {
            color: variables.btnDangerBg,
          },
        },
        backgroundColor: variables.btnDangerBg,
      },

      '.block': {
        justifyContent: 'center',
        alignSelf: 'stretch',
      },

      '.full': {
        justifyContent: 'center',
        alignSelf: 'stretch',
        borderRadius: 0
      },

      '.rounded': {
        paddingHorizontal: variables.buttonPadding + 20,
        borderRadius: variables.borderRadiusLarge,
      },


      '.transparent': {
        backgroundColor: 'transparent',
        elevation: 0,
        shadowColor: null,
        shadowOffset: null,
        shadowRadius: null,
        shadowOpacity: null,

        'NativeBase.Text': {
          color: variables.btnPrimaryBg,
        },
        'NativeBase.Icon': {
          color: variables.btnPrimaryBg,
        },
        'NativeBase.IconNB': {
          color: variables.btnPrimaryBg,
        },
        '.dark': {
          'NativeBase.Text': {
            color: '#000',
          },
          'NativeBase.IconNB': {
            color: '#000',
          },
          'NativeBase.Icon': {
            color: '#000',
          },
          backgroundColor: null,
        },
        '.danger': {
          'NativeBase.Text': {
            color: variables.btnDangerBg,
          },
          'NativeBase.IconNB': {
            color: variables.btnDangerBg,
          },
          'NativeBase.Icon': {
            color: variables.btnDangerBg,
          },
          backgroundColor: null,
        },
        '.warning': {
          'NativeBase.Text': {
            color: variables.btnWarningBg,
          },
          'NativeBase.IconNB': {
            color: variables.btnWarningBg,
          },
          'NativeBase.Icon': {
            color: variables.btnWarningBg,
          },
          backgroundColor: null,
        },
        '.info': {
          'NativeBase.Text': {
            color: variables.btnInfoBg,
          },
          'NativeBase.IconNB': {
            color: variables.btnInfoBg,
          },
          'NativeBase.Icon': {
            color: variables.btnInfoBg,
          },
          backgroundColor: null,
        },
        '.primary': {
          'NativeBase.Text': {
            color: variables.btnPrimaryBg,
          },
          'NativeBase.IconNB': {
            color: variables.btnPrimaryBg,
          },
          'NativeBase.Icon': {
            color: variables.btnPrimaryBg,
          },
          backgroundColor: null,
        },
        '.success': {
          'NativeBase.Text': {
            color: variables.btnSuccessBg,
          },
          'NativeBase.IconNB': {
            color: variables.btnSuccessBg,
          },
          'NativeBase.Icon': {
            color: variables.btnSuccessBg,
          },
          backgroundColor: null,
        },
        '.light': {
          'NativeBase.Text': {
            color: '#f4f4f4',
          },
          'NativeBase.IconNB': {
            color: '#f4f4f4',
          },
          'NativeBase.Icon': {
            color: '#f4f4f4',
          },
          backgroundColor: null,
        },


      },

      '.small': {
        height: 30,
        'NativeBase.Text': {
          fontSize: 14,
        },
      },

      '.large': {
        height: 60,
        'NativeBase.Text': {
          fontSize: 22,
          lineHeight: 24,
        },
      },

      '.iconRight': {
        'NativeBase.Text': {
          marginRight: variables.buttonPadding,
        },
      },
      '.iconLeft': {
        'NativeBase.Text': {
          marginLeft: variables.buttonPadding,
        },
      },


      '.capitalize': {

      },

      '.vertical': {
        flexDirection: 'column',
        height: null,
      },


      'NativeBase.Text': {
        fontFamily: variables.btnFontFamily,
        marginLeft: 0,
        marginRight: 0,
        color: variables.inverseTextColor,
        fontSize: variables.btnTextSize,
        lineHeight: variables.btnLineHeight,
        // childPosition: 1
      },

      'NativeBase.Icon': {
        color: variables.inverseTextColor,
        fontSize: 24,
        marginHorizontal: 5,
        paddingTop: (platform === 'ios') ? 2 : undefined
      },
      'NativeBase.IconNB': {
        color: variables.inverseTextColor,
        fontSize: 24,
        marginHorizontal: 5,
        paddingTop: (platform === 'ios') ? 2 : undefined
      },
      '.iconLeft': {
        'NativeBase.IconNB': {
          marginRight: 10,
          marginLeft: 0,
        },
        'NativeBase.Icon': {
          marginRight: 10,
          marginLeft: 0,
        },
      },
      '.iconRight': {
        'NativeBase.IconNB': {
          marginLeft: 10,
          marginRight: 0,
        },
        'NativeBase.Icon': {
          marginLeft: 10,
          marginRight: 0,
        },
      },
      '.picker': {
        'NativeBase.Text': {
          '.note': {
            fontSize: 16,
            lineHeight: null,
          },
        },
      },

      paddingVertical: variables.buttonPadding,
      paddingHorizontal: variables.buttonPadding + 10,
      backgroundColor: variables.btnPrimaryBg,
      borderRadius: variables.borderRadiusBase,
      borderColor: variables.btnPrimaryBg,
      borderWidth: null,
      height: 45,
      alignSelf: 'flex-start',
      flexDirection: 'row',
      elevation: 2,
      shadowColor: (platformStyle==='material') ? '#000' : undefined,
      shadowOffset: (platformStyle==='material') ? {width: 0, height: 2} : undefined,
      shadowOpacity: (platformStyle==='material') ?  0.2 : undefined,
      shadowRadius: (platformStyle==='material') ? 1.2 : undefined,
      alignItems: 'center',
      justifyContent: 'space-between',
  };
  return buttonTheme;
};
