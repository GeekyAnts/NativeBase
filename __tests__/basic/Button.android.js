import 'react-native';
import renderer from 'react-test-renderer';

import React from 'react';
import { Button } from './../../src/basic/Button';
// import Icon from '../../../Components/Widgets/Icon';

// Note: test renderer must be required after react-native.

jest.mock('Platform', () => {
    const Platform = require.requireActual('Platform');
    Platform.OS = 'android';
    return Platform;
});


it('renders regular button', () => {
   const tree = renderer.create(
       <Button>
           Test
       </Button>
   ).toJSON();
    expect(tree).toMatchSnapshot();
});