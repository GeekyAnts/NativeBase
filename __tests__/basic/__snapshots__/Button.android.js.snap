exports[`test renders regular button 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 2,
      "opacity": 1,
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "alignSelf": "center",
      }
    } />
  Test
</View>
`;
